service/src/main/java/bio/terra/lz/futureservice/common/utils/RequestQueryParamUtils.java

```diff
@@ -0,0 +1,32 @@
+package bio.terra.lz.futureservice.common.utils;
+
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.UUID;
+import org.apache.commons.lang3.StringUtils;
+
+public class RequestQueryParamUtils {
+  private static final String BILLING_PROFILE_ID_PARAM_NAME = "billingProfileId";
+
+  private RequestQueryParamUtils() {}
+
+  public static boolean isBillingProfileIdSanitized(
+      UUID billingProfileId, String requestQueryParamPairs) {
+    if (requestQueryParamPairs == null) return false;
+
+    // unlikely, but we might have more than 1 parameter provided;
+    // billingProfileId=test&param1=value1
+    String[] allQueryParamsPairs = requestQueryParamPairs.split("&");
+    Optional<String> billingProfileIdQueryParamPair =
+        Arrays.stream(allQueryParamsPairs)
+            .filter(pairs -> pairs.contains(BILLING_PROFILE_ID_PARAM_NAME))
+            .findFirst();
+
+    if (billingProfileIdQueryParamPair.isEmpty()) return false;
+
+    String[] billingProfileIdKeyValuePair = billingProfileIdQueryParamPair.get().split("=");
```

Comment:
I believe there is a `getParameterMap` method that would allow you to avoid having to parse them yourselves.

Reply:
(instead of using `request.getQueryString`)

Reply:
Thanks. That's good observation. I've just checked and only getQueryString() returns original raw data.
![image](https://github.com/DataBiosphere/terra-landing-zone-service/assets/91149690/6afc45d6-5b6d-404b-a79e-b7904c7c43b0)

Reply:
Ugh, that's frustrating. Thanks for checking!